src/Compilers/Core/Portable/Operations/OperationFactory.cs

```diff
@@ -81,6 +81,7 @@ public static IInvalidOperation CreateInvalidExpression(SemanticModel semanticMo
             return new InvalidOperation(children, semanticModel, syntax, type: null, constantValue: default(Optional<object>), isImplicit: isImplicit);
         }
 
+        public static Lazy<IOperation> EmptyOperation { get; } = new Lazy<IOperation>(() => null);
```

Comment:
>public static Lazy<IOperation> EmptyOperation { get; } = new Lazy<IOperation>(() => null); [](start = 8, length = 90)

Both this field and the next one should probably be readonly.

Reply:
Never mind these are properties. The question is why?

---
In reply to: [149577275](https://github.com/dotnet/roslyn/pull/22972#discussion_r149577275) [](ancestors = 149577275)

Reply:
We have an open bug tracking making these use the same technique used in workspaces, [https://github.com/dotnet/roslyn/issues/22938](https://github.com/dotnet/roslyn/issues/22938), so I'm not going to change the implementation of these in this PR. I will change the names and remove the _lazyNothingOperation fields.

---
In reply to: [149577531](https://github.com/dotnet/roslyn/pull/22972#discussion_r149577531) [](ancestors = 149577531,149577275)